src/Symfony/Component/Messenger/Middleware/HandleMessageMiddleware.php

```diff
@@ -97,6 +106,14 @@ public function handle(Envelope $envelope, StackInterface $stack): Envelope
             } catch (\Throwable $e) {
                 $exceptions[$handlerDescriptor->getName()] = $e;
```

Comment:
```suggestion
                $exceptions[$handlerDescriptor->getName()] = $e;
                
                $this->eventDispatcher?->dispatch(new HandlerFailureEvent($envelope, $handlerDescriptor->getName(), $e));
```

Same for the success case and remove the `$e` variable?

Reply:
I don't mind dispatching the failure event in the `catch` block, however I'm a bit reluctant about dispatching the success event in the `try` block, as a failure in any listener/subscriber of this event would mark the *handler* as failed.

 That's why I kept things outside the `try/catch`.